docs/raw/push-notification-server.md

```diff
@@ -0,0 +1,766 @@
+
+If `token`,`installation_id`,`device_tokens`,`version` are empty, a response MUST
+be sent with `error` set to `MALFORMED_MESSAGE`.
+
+If the `version` is equal or less than the currently stored version, a response MUST
+be sent with `error` set to `VERSION_MISMATCH`.
+
+If any other error occurs the `error` should be set to `INTERNAL_ERROR`.
+
+If the request is successful, an access token MUST be generated and sent back to the client
+with `success` set to `true`.
+
+Otherwise a response MUST be sent with `success` set to `false`.
+
+It is RECOMMENDED to use a randomly generated string for `access_token` of at least 36 characters.
+
+`request_id` should be set to the `SHA3-256` of the signature sent by the client.
+
+The request MUST be sent on the [partitioned topic][./2-account.md] of the sender.
+
+The payload of the response is:
+
+```protobuf
+message PushNotificationPreferencesResponse {
+  boolean success = 1;
+  ErrorType error = 2;
+  bytes request_id = 3;
+  PushNotificationPreferences preferences = 4;
+  string access_token = 5;
+
+  enum ErrorType {
+    UNKNOWN_ERROR_TYPE = 0;
+    MALFORMED_MESSAGE = 1;
+    VERSION_MISMATCH = 2;
+    UNSUPPORTED_TOKEN_TYPE = 3;
+    INTERNAL_ERROR = 4;
+  }
+}
+```
+
+[//]: (We can ratched and use a partitioned topic here)
+
+A client SHOULD listen for a response sent on their [partitioned topic][./2-account.md].
+
+If `success` is `true` the client has registered successfully.
+
+If `success` is `false`:
+
+- If `MALFORMED_MESSAGE` is returned, the request SHOULD not be retried without ensuring 
+  that is correctly formed.
+- If `INTERNAL_ERROR` is returned, the request MAY be retried, but the client MUST be 
+  backoff exponentially
+- If `VERSION_MISMATCH` is returned, the client SHOULD ensure that the remote version returned in `preferences` is integrated, and the version should be incremented and the request MAY be sent again.
+
+A client MAY register with multiple Push Notification Servers in order to increase availability.
+
+A client SHOULD make sure that all the notification services they registered with have the same information about their devices and tokens.
+
+If no response is returned the request SHOULD be considered failed and MAY be retried with the same server or a different one, but it MUST exponentially backoff after each trial.
+
+If the request is successful a token is returned. This SHOULD be 
+[advertised](#advertising-a-push-notification-server) as described below if no public key filtering is necessary.
+
+## Re-registering with the push notification server
+
+A client SHOULD re-register with the node if the APN or FIREBASE token changes.
+
+When re-registering a client SHOULD ensure that it has the most up-to-date
+`PushNotificationPreferences`, update the part relative to their `installation_id`
+if necessary, increment `version` and send a `PushNotificationRegister` as described above.
+
+Once re-registered, a client SHOULD advertise the changes.
```

Comment:
Does the node need special behaviour for re-registering, or will the previously given registration details be overwritten and behave exactly the same as a standard registration?

Reply:
No special behavior, the previously given registration details will be overwritten.

Reply:
Correct